src/dndbeyond/content-scripts/character.js

```diff
@@ -997,58 +998,124 @@ function rollAction(paneClass, force_to_hit_only = false, force_damages_only = f
-        for (let i = 0; i < damages.length; i++){
-            if ((damage_types[i] === "Acid" || damage_types[i] === "Fire" || damage_types[i] === "Necrotic" || damage_types[i] === "Poison") &&
-                alchemical_savant_regex.test(damages[i])) {
-                damages.push(`${character.getAbility("INT").mod < 2 ? 1 : character.getAbility("INT").mod}`);
-                damage_types.push("Alchemical Savant");
-                break;
+    
+    if (character.hasClass("Druid")) {
+        // Druid: Wildfire Druid: Enhanced Bond
+        if (character.hasClassFeature("Enhanced Bond") &&
+            character.getSetting("wildfire-spirit-enhanced-bond", false) &&
+            damages.length > 0) {
+            for (let i = 0; i < damages.length; i++){
+                if (damage_types[i] === "Fire") {
+                    damages.push("1d8");
+                    damage_types.push("Enhanced Bond");
+                    break;
+                }
             }
         }
     }
-    
-    //Cleric Blessed Strikes
-    if (character.hasClassFeature("Blessed Strikes") &&
-        character.getSetting("cleric-blessed-strikes", false) &&
-        spell_level.includes("Cantrip")) {
-        damages.push("1d8");
-        damage_types.push("Blessed Strikes");
+
+    if (character.hasClass("Ranger")) {
+        // Ranger: Favored Foe
+        if (character.hasClassFeature("Favored Foe") &&
+            character.getSetting("ranger-favored-foe", false) &&
+            to_hit !== null) {
+            const ranger_level = character.getClassLevel("Ranger");
+            damages.push(ranger_level < 6 ? "1d4" : ( ranger_level < 14 ? "1d6" : "1d8"));
+            damage_types.push("Favored Foe");
+        }
+        
+        // Ranger: Gathered Swarm
+        if (character.hasClassFeature("Gathered Swarm") &&
+            character.getSetting("ranger-gathered-swarm", false) &&
+            to_hit !== null) {
+            const ranger_level = character.getClassLevel("Ranger");
+            damages.push(ranger_level < 11 ? "1d6" : "1d8");
+            damage_types.push("Gathered Swarm");
+        }
     }
 
-    if (character.hasClassFeature("Enhanced Bond") &&
-        character.getSetting("wildfire-spirit-enhanced-bond", false) &&
-        damages.length > 0) {
-        for (let i = 0; i < damages.length; i++){
-            if (damage_types[i] === "Fire") {
-                damages.push("1d8");
-                damage_types.push("Enhanced Bond");
-                break;
-            }
+    if (character.hasClass("Warlock")) {
+        // Warlock: The Hexblade: Hexblade's Curse
+        if (damages.length > 0 &&
+            character.getSetting("warlock-hexblade-curse", false) &&
+            character.hasClassFeature("Hexblade’s Curse") &&
+            character._proficiency !== null) {
+            damages.push(character._proficiency);
+            damage_types.push("Hexblade's Curse");
+        }
```

Comment:
You can't have hexblade's curse here. It's already in the generic handler, so a warlock doing a melee attack with green flame blade means it gets added twice.